### PR TITLE
Reimplement extension methods using existing combinators

### PR DESCRIPTION
**Pull request by bash**

### CODE REVIEW

File: Funcky.Async/Extensions/AsyncEnumerableExtensions/FirstOrNone.cs [hunk 1]
```diff
@@ -14,38 +14,28 @@ public static partial class AsyncEnumerableExtensions
...
-            {
-                if (await predicate(item, cancellationToken))
-                {
-                    return item;
-                }
-            }
-
-            return Option<TSource>.None();
-        }
+            => await source.WhereAwaitWithCancellation(predicate).Select(Option.Some).FirstOrDefaultAsync(cancellationToken);
```
Comment (FreeApophis) [inline Funcky.Async/Extensions/AsyncEnumerableExtensions/FirstOrNone.cs:39]: Still no fully understand all the Postfixes. However in this case you use a Function "WithCancellation" but your function does "WithCancellationAsync" - it looks like this is a mismatch. Is the naming wrong? Is the wrong method called?

File: Funcky.Async/Extensions/AsyncEnumerableExtensions/LastOrNone.cs [hunk 2]
```diff
@@ -14,40 +14,28 @@ public static partial class AsyncEnumerableExtensions
...
-            {
-                if (await predicate(item, cancellationToken))
-                {
-                    lastItem = item;
-                }
-            }
-
-            return lastItem;
-        }
+            => await source.WhereAwaitWithCancellation(predicate).Select(Option.Some).LastOrDefaultAsync(cancellationToken);
```
Comment (FreeApophis) [inline Funcky.Async/Extensions/AsyncEnumerableExtensions/LastOrNone.cs:39]: Still no fully understand all the Postfixes. However in this case you use a Function "WithCancellation" but your function does "WithCancellationAsync" - it looks like this is a mismatch. Is the naming wrong? Is the wrong method called?

File: Funcky.Async/Extensions/AsyncEnumerableExtensions/Pairwise.cs [hunk 3]
```diff
@@ -32,8 +31,8 @@ public static IAsyncEnumerable<(TSource Front, TSource Back)> Pairwise<TSource>(
             => AsyncEnumerable.Create(cancellationToken => PairwiseInternal(source, resultSelector, cancellationToken));
 
         [Pure]
-#pragma warning disable 8425
-        public static async IAsyncEnumerator<TResult> PairwiseInternal<TSource, TResult>(this IAsyncEnumerable<TSource> source, Func<TSource, TSource, TResult> resultSelector, CancellationToken cancellationToken)
+        #pragma warning disable 8425
```
Comment (FreeApophis) [inline Funcky.Async/Extensions/AsyncEnumerableExtensions/Pairwise.cs:34]: Why is here a warnig disable? Removing it, does not issue a warning on my side.

**Review by FreeApophis — CHANGES_REQUESTED** [review]

File: Funcky.Async/Extensions/AsyncEnumerableExtensions/Pairwise.cs (hunk 3 above)
Comment (bash) [reply Funcky.Async/Extensions/AsyncEnumerableExtensions/Pairwise.cs]: 524ec662cce812946cc9473e55b67f3c06c9cc77

**Review by FreeApophis — APPROVED** [review]

**Pull request merged**